models/git/commit.go

```diff
@@ -0,0 +1,542 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+package git
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"html/template"
+	"io"
+	"slices"
+	"strconv"
+
+	"code.gitea.io/gitea/models/db"
+	"code.gitea.io/gitea/models/renderhelper"
+	repo_model "code.gitea.io/gitea/models/repo"
+	user_model "code.gitea.io/gitea/models/user"
+	"code.gitea.io/gitea/modules/json"
+	"code.gitea.io/gitea/modules/markup/markdown"
+	"code.gitea.io/gitea/modules/setting"
+	"code.gitea.io/gitea/modules/storage"
+	"code.gitea.io/gitea/modules/timeutil"
+	"github.com/google/uuid"
+)
+
+type (
+	CommitDataList []*CommitData
+	ReactionMap    map[string][]string
+	AttachmentMap  map[string]*AttachmentOptions
+)
+
+type CommitData struct {
+	ID              int64            `xorm:"pk autoincr"`
+	PosterID        int64            `xorm:"INDEX"`
+	Poster          *user_model.User `xorm:"-"`
+	Line            int64
+	FileName        string
+	CommitSHA       string                 `xorm:"VARCHAR(64)"`
+	Attachments     string                 `xorm:"JSON TEXT"`
+	Reactions       string                 `xorm:"JSON TEXT"`
+	Comment         string                 `xorm:"LONGTEXT"`
+	RenderedComment template.HTML          `xorm:"-"`
+	ContentVersion  int                    `xorm:"NOT NULL DEFAULT 0"`
+	RefRepoID       int64                  `xorm:"index"`
+	Repo            *repo_model.Repository `xorm:"-"`
+	CreatedUnix     timeutil.TimeStamp     `xorm:"INDEX created"`
+	UpdatedUnix     timeutil.TimeStamp     `xorm:"INDEX updated"`
+}
```

Comment:
It's better there is a reply-to column so that we can have a tree view for the conversation.